src/pip/_internal/utils/misc.py

```diff
@@ -1081,6 +1081,25 @@ def path_to_url(path):
     return url
 
 
+def build_url_from_netloc(netloc, scheme='https'):
+    """
+    Build a full URL from a netloc.
+    """
+    if netloc.count(':') >= 2 and '[' not in netloc:
+        # It must be a bare IPv6 address, then wrap it with brackets.
+        netloc = '[{}]'.format(netloc)
+    return '{}://{}'.format(scheme, netloc)
+
+
+def does_netloc_has_port(netloc):
```

Comment:
`does_netloc_have_port()` would be the way to say it.

Reply:
I would like to rename to `netloc_has_port()`, which looks more natural in
`if netloc_has_port(netloc)`

Reply:
Okay.